src/hook/pauseResume.js

```diff
@@ -0,0 +1,299 @@
+/**
+* Copyright (c) 2015 NAVER Corp.
+* egjs projects are licensed under the MIT license
+*/
+eg.module("pauseResume", ["jQuery"], function($) {
+	"use strict";
+
+	var animateFn = $.fn.animate;
+	var stopFn = $.fn.stop;
+	var uuid = 1;
+
+	function now() {
+		return new Date().getTime();
+	}
+
+	function AniProperty(prop, optall, prevProp, isFirst) {
+		this.opt = optall;
+		this.start = -1;
+		this.elapsed = 0;
+		this.paused = false;
+		this.uuid = uuid++;
+		this.easingNames = [];
+		this.prop = isFirst ? {} : $.extend({}, prevProp);
+
+		for (var propName in prop) {
+			var propValue = prop[propName];
+			var markIndex;
+			var sign;
+
+			//If it has a absoulte value.
+			if (typeof(propValue) !== "string" ||
+				(markIndex = propValue.search(/[+|-]=/)) < 0) {
+				this.prop[propName] = propValue;
+				continue;
+			}
+
+			sign = propValue.charAt(markIndex) === "-" ? -1 : 1;
+
+			//If it has a relative value
+			this.prop[propName] = propValue
+				.replace(/([-|+])*([\d|\.])+/g,
+					generateAbsoluteValMaker(prevProp, propName, sign))
+				.replace(/[-|+]+=/g, "");
+		}
+	}
+
+	/**
+	 * Generate a new absolute value maker.
+	 *
+	 * function to avoid JS Hint error "Don't make functions within a loop"
+	 */
+	function generateAbsoluteValMaker(prevProp, propName, sign) {
+		return function absoluteValMaker(match) {
+			var currValue = prevProp[propName] ? parseFloat(prevProp[propName]) : 0;
+			return currValue + Number(match) * sign;
+		};
+	}
+
+	AniProperty.prototype.init = function() {
+		this.start = now();
+		this.elapsed = 0;
+	};
+
+	AniProperty.prototype.addEasingFn = function(easingName) {
+		this.easingNames.push(easingName);
+	};
+
+	AniProperty.prototype.clearEasingFn = function() {
+		var easing;
+		while (easing = this.easingNames.shift()) {
+			delete $.easing[easing];
+		}
+		this.easingNames = [];
+	};
+
+	function addAniProperty(el, prop, optall) {
+		var prevProp;
+		var propCount = el.__aniProps ? el.__aniProps.length : 0;
+		var isFirst = false;
+		var newProp;
+
+		if (propCount === 0) {
+			prevProp = el.style;
+			isFirst = true;
+		} else {
+			prevProp = el.__aniProps[propCount - 1].prop;
+		}
+
+		//prevProp is used for calculating absolute value by accumulating aniProps.
+		//So newProp has absolute value accumlated.
+		newProp = new AniProperty(prop, optall, prevProp, isFirst);
+		el.__aniProps = el.__aniProps || [];
+
+		//Animation is excuted immediately.
+		if (el.__aniProps.length === 0) {
+			newProp.init();
+		}
+		el.__aniProps.push(newProp);
+	}
+
+	function getOptAll(speed, easing, fn) {
+		var opt = speed && typeof speed === "object" ?
+		jQuery.extend({}, speed) : {
+			complete: fn || !fn && easing ||
```

Comment:
@mixed I'll check it more. This parameter processing is brought from jQuery.speed.